### PR TITLE
added procedure_harness

### PR DESCRIPTION
**Pull request by lucadealfaro**
I made this PR so I can comment on it.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## Readme
 
+
```
Comment (lucadealfaro) [inline README.markdown]: This is the only line that changed.  Perhaps you forgot to add files?

Comment (ahume128) [inline README.markdown]: No sorry I was just trying to push the branch so both team members could work on it and at first I thought I needed to make a change to do so.

Comment (lucadealfaro) [inline README.markdown]: Ah ok :-) Sorry; forgetting to add files is such a common error (I'd do it more were it not for the fact that pycharm asks me whenever I create a file whether I want to add it to git).

File: applications/welcome/models/tables.py [hunk 2]
```diff
@@ -17,3 +17,13 @@
 
 ## after defining tables, uncomment below to enable auditing
 # auth.enable_record_versioning(db)
+
+db.define_table('procedures',
+                Field('procedure_id', 'bigint', required=True), #key
```
Comment (lucadealfaro) [inline applications/welcome/models/tables.py]: Automatic from web2py.

File: applications/welcome/models/tables.py [hunk 3]
```diff
@@ -17,3 +17,13 @@
...
+                Field('device_id', required=True), # not sure what field type this should be
```
Comment (lucadealfaro) [inline applications/welcome/models/tables.py]: Read from a settings table that the device user management team is doing.

File: applications/welcome/models/proc_harness.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#########################################################################
+## Define API for interacting with procedure table
+## These methods will be used in the controller, called by the code
+## that actually performs the sync to interact with the tables
+#########################################################################
+
+from gluon import current
+
+proc_table = current.db["procedures"]
+
```
Comment (lucadealfaro) [inline applications/welcome/models/proc_harness.py]: def do_synch():
    """Ask the server for all the procedures for this device_id, and get the latest version of each.  Write it in the proper files and update the tables."""

**Comment by lucadealfaro** [comment]
This PR seems to need more work before it can be merged.

**Pull request closed without merging**